packages/synapse-react-client/src/components/ChangePassword/ChangePassword.integration.test.tsx

```diff
@@ -0,0 +1,440 @@
+      expect(newPasswordField).not.toBeInTheDocument()
+      expect(confirmPasswordField).not.toBeInTheDocument()
+
+      expect(changePasswordSpy).toHaveBeenCalledTimes(1)
+      expect(changePasswordSpy).toHaveBeenCalledWith({
+        concreteType:
+          'org.sagebionetworks.repo.model.auth.ChangePasswordWithCurrentPassword',
+        username: mockUserProfileData.userName,
+        newPassword: newPassword,
+        currentPassword: currentPassword,
+      })
+    })
+
+    // Type the code. Once the code is entered, the form should submit automatically
+    for (let i = 0; i < otpInputs.length; i++) {
+      await userEvent.type(otpInputs[i], String(i + 1))
+    }
+    await waitFor(() => {
+      expect(changePasswordSpy).toHaveBeenCalledTimes(2)
+      expect(changePasswordSpy).toHaveBeenLastCalledWith({
+        concreteType:
+          'org.sagebionetworks.repo.model.auth.ChangePasswordWithTwoFactorAuthToken',
+        newPassword: newPassword,
+        twoFaToken: twoFaToken,
+        userId: userId,
+        otpCode: '123456',
+        otpType: 'TOTP',
+      })
+
+      expect(mockDisplayToast).toHaveBeenCalledWith(
+        'Password successfully changed.',
+        'success',
+      )
+    })
+
+    // The form should be reset
+    expect(getCurrentPasswordField()).toHaveValue('')
+    expect(getNewPasswordField()).toHaveValue('')
+    expect(getConfirmPasswordField()).toHaveValue('')
+  })
+
+  it('supports entering a recovery code if the server requires 2FA', async () => {
+    const userId = MOCK_USER_ID
+    const twoFaToken = 'mock-2fa-token'
+    server.use(
+      // The first call will indicate that 2FA is required
+      getRequires2FAChangePasswordHandler(
+        getEndpoint(BackendDestinationEnum.REPO_ENDPOINT),
+        userId,
+        twoFaToken,
+      ),
+      // Update the mock server so the second request will succeed
+      getSuccessfulChangePasswordHandler(
+        getEndpoint(BackendDestinationEnum.REPO_ENDPOINT),
+      ),
+    )
+
+    const currentPassword = 'currentPassword'
+    const newPassword = 'newPassword'
+
+    const {
+      user,
+      currentPasswordField,
+      newPasswordField,
+      confirmPasswordField,
+      submitButton,
+    } = setUp()
+
+    await user.type(currentPasswordField, currentPassword)
+    await user.type(newPasswordField, newPassword)
+    await user.type(confirmPasswordField, newPassword)
+
+    await userEvent.click(submitButton)
+
+    const useRecoveryCodeButton = await screen.findByText(
+      'Use a backup code instead',
+    )
+    await userEvent.click(useRecoveryCodeButton)
+
+    const recoveryCode = '1234-5678-abcd-edcb'
+    const recoveryCodeTextbox = await screen.findByPlaceholderText(
+      'Enter backup code',
+    )
+    await user.type(recoveryCodeTextbox, recoveryCode)
+    await user.click(screen.getByRole('button', { name: 'Submit' }))
+
+    // Type the code. Once the code is entered, the form should submit automatically
```

Comment:
Looks like this comment may refer to the TOTP form, since the recovery code form requires the user to click submit. Should the comment be removed or updated?